metadata/metadata.go

```diff
@@ -141,6 +141,8 @@ func NewContext(ctx context.Context, md MD) context.Context {
 }
 
 // FromContext returns the MD in ctx if it exists.
+// The returned md is not thread safe, writing to it may cause concurrency issues.
```

Comment:
don't use "thread safe".  have sth like

"md cannot be accessed by multiple goroutines  concurrently."